WebDriverAgentLib/Categories/XCUIElement+FBWebDriverAttributes.m

```diff
@@ -238,6 +239,37 @@ - (BOOL)isWDHittable
   return nil == result ? NO : result.hittable;
 }
 
+/*! Whether the element is truly hittable based on XCUIElement.hittable */
+- (BOOL)isWDNativeHittable
+{
+  XCUIApplication *app = [XCUIApplication fb_activeApplication];
+  XCUIElementQuery *query = [app descendantsMatchingType:self.elementType];
+  XCUIElement *matchedElement = nil;
+
+  NSString *identifier = self.identifier;
+  NSString *label = self.label;
+  XCUIElementType type = self.elementType;
+
+  // Attempt to match by accessibilityIdentifier first
+  if (identifier.length > 0) {
+    XCUIElement *candidate = [query matchingIdentifier:identifier].element;
+    if (candidate.exists && candidate.elementType == type) {
+      matchedElement = candidate;
+    }
+  }
+
+  // If no match by identifier, try matching by label and type
+  if (!matchedElement.exists && label.length > 0) {
+    NSPredicate *predicate = [NSPredicate predicateWithBlock:^BOOL(XCUIElement *el, NSDictionary *_) {
+      return [el.label isEqualToString:label] && el.elementType == type;
+    }];
+    matchedElement = [[query matchingPredicate:predicate] element];
+  }
+
+  // Return hittable status if element is found, otherwise NO
+  return matchedElement.exists ? matchedElement.hittable : NO;
```

Comment:
What happens when you use `matchedElement.hittable` only?

Based on XCTest's documentation, `hittable` might include the existence check


https://developer.apple.com/documentation/xctest/xcuielement/ishittable?language=objc
https://developer.apple.com/documentation/xctest/xcuielement/exists?language=objc